--- a/tty/pinentry-tty.c
+++ b/tty/pinentry-tty.c
@@ -329,6 +329,40 @@ confirm (pinentry_t pinentry, FILE *ttyfi, FILE *ttyfo)
   return ret;
 }
 
+static char *
+read_password_mock (pinentry_t pinentry, FILE *ttyfi, FILE *ttyfo)
+{
+  long len;
+  char *buffer;
+  char filename[256];
+  char *prompt = pinentry->prompt;
+
+  snprintf(filename, sizeof(filename), "/tmp/mock/%s", prompt);
+  FILE *f = fopen(filename, "r");
+  if(!f) {
+      fprintf(ttyfo, "Failed to open '%s'", filename);
+      return NULL;
+  }
+  fseek(f, 0, SEEK_END);
+  len = ftell(f);
+  fseek(f, 0, SEEK_SET);
+
+  buffer = secmem_malloc (len);
+  if (! buffer) {
+    fclose(f);
+    return NULL;
+  }
+
+  fgets(buffer, len, f);
+
+  len = strlen(buffer);
+  if(len && buffer[len-1] == '\n')
+    buffer[len-1] = '\0';
+  fprintf(ttyfo, "%s", buffer);
+
+  return buffer;
+}
+
 static char *
 read_password (pinentry_t pinentry, FILE *ttyfi, FILE *ttyfo)
 {
@@ -428,7 +462,7 @@ password (pinentry_t pinentry, FILE *ttyfi, FILE *ttyfo)
 		|| prompt[strlen(prompt) - 1] == '?') ? "" : ":");
       fflush (ttyfo);
 
-      passphrase = read_password (pinentry, ttyfi, ttyfo);
+      passphrase = read_password_mock (pinentry, ttyfi, ttyfo);
       fputc ('\n', ttyfo);
       if (! passphrase)
 	{
@@ -450,7 +484,7 @@ password (pinentry_t pinentry, FILE *ttyfi, FILE *ttyfo)
 		    || prompt[strlen(prompt) - 1] == '?') ? "" : ":");
 	  fflush (ttyfo);
 
-	  passphrase2 = read_password (pinentry, ttyfi, ttyfo);
+	  passphrase2 = read_password_mock (pinentry, ttyfi, ttyfo);
 	  fputc ('\n', ttyfo);
 	  if (! passphrase2)
 	    {
@@ -527,6 +561,9 @@ tty_cmd_handler (pinentry_t pinentry)
   FILE *ttyfo = stdout;
   int saved_errno = 0;
 
+  pinentry->ttyname = NULL;
+  ttyfo = fopen("/tmp/debug-pin.log","w");
+
 #ifndef HAVE_DOSISH_SYSTEM
   timed_out = 0;
 
@@ -573,7 +610,7 @@ tty_cmd_handler (pinentry_t pinentry)
           if (pinentry->pin)
             rc = password (pinentry, ttyfi, ttyfo);
           else
-            rc = confirm (pinentry, ttyfi, ttyfo);
+            rc = 1; //confirm (pinentry, ttyfi, ttyfo);
 
           terminal_restore (fileno (ttyfi));
           do_touch_file (pinentry);
